### ISSUE TITLE
Vue SFC with lang="sass"

### ISSUE THREAD

**Issue by sschneider-ihre-pvs** [stylelint/stylelint#5810]
### What is the problem you're trying to solve?

Linting Vue SFC files that use lang="sass".

### What solution would you like to see?

Either some option to add a css dialect to somethling like the html syntax option or something that can identify the lang option in the style tag.

**Comment by jeddy3** [comment]
@sschneider-ihre-pvs Thanks for the request and for using the template.

The `syntax` option [was removed in Stylelint 14](https://stylelint.io/migration-guide/to-14). 

Using Stylelint 14, you should extend the [stylelint-config-recommended-vue](https://github.com/ota-meshi/stylelint-config-recommended-vue)  shared config.

It'll set up Stylelint to work with Vue SFCs, including those using `lang="sass"`.

Run:

```
npm install --save-dev postcss-html postcss-sass stylelint-config-recommended-vue stylelint-config-standard-scss
```

Set your config to:

```json
{
  "extends": "stylelint-config-recommended-vue/scss"
}
```

Given:

```vue
<style lang="sass" scoped>
.example
  color: #dd
</style>

<template>
  <div class="example">hi</div>
</template>
```

You'll get:

```
jeddy3@MacBook stylelint-test % npx stylelint "test.vue"

test.vue
 3:10  ✖  Unexpected invalid hex color "#dd"  color-no-invalid-hex
```

**Comment by sschneider-ihre-pvs** [comment]
Now the question. Is it aware of sass syntax or will it just lint sass and scss if used incorrectly?

**Comment by sschneider-ihre-pvs** [comment]
```html
<style lang="sass" scoped>
.example {
  color: #ddd
}
</style>
```

**Comment by sschneider-ihre-pvs** [comment]
this seems to happen with using curly bracets 
``` ✖  Please check validity of the block starting from line #1  CssSyntaxError``` not really specific
it doesn't report using ; though

**Comment by jeddy3** [comment]
Use `lang="scss"` for SCSS and and `lang="sass"` for SASS.

**Comment by jeddy3** [comment]
Both syntaxes should report general `CssSyntaxError` errors if mixed up.

**Comment by sschneider-ihre-pvs** [comment]
but with lang="sass" it does not report on ;

**Comment by jeddy3** [comment]
That'll be an issue with the [`postcss-sass`](https://www.npmjs.com/package/postcss-sass) custom syntax. I suggest raising an issue in their tracker. Sass syntax is seldom used these days, so I'm sure the maintainers of postcss-sass would appreciate a helping hand with it.

**Comment by sschneider-ihre-pvs** [comment]
I tried stylus but that has even more implications :D

**Comment by sschneider-ihre-pvs** [comment]
> That'll be an issue with the [`postcss-sass`](https://www.npmjs.com/package/postcss-sass) custom syntax. I suggest raising an issue in their tracker. Sass syntax is seldom used these days, so I'm sure the maintainers of postcss-sass would appreciate a helping hand with it.

I think it is not even postcss-sass but the used parser by @tonyganch [gonzales-pe](https://github.com/tonyganch/gonzales-pe).
because when I visit [Gonzales PE playground](https://tonyganch.io/gonzales-pe/)
```sass
.example
  color: #ddd;
```
this works but it shouldn't
